hadoop-hdds/client/src/main/java/org/apache/hadoop/hdds/scm/storage/BlockDataStreamOutput.java

```diff
@@ -184,7 +186,47 @@ private DataStreamOutput setupStream() throws IOException {
         ContainerCommandRequestMessage.toMessage(builder.build(), null);
 
     return Preconditions.checkNotNull(xceiverClient.getDataStreamApi())
-        .stream(message.getContent().asReadOnlyByteBuffer());
+//        .stream(message.getContent().asReadOnlyByteBuffer());
+    .stream(message.getContent().asReadOnlyByteBuffer(),getRoutingTable(pipeline));
+  }
+
+  public RoutingTable getRoutingTable(Pipeline pipeline) {
+    RaftPeer primary = null;
+    List<RaftPeer> raftPeers = new ArrayList();
```

Comment:
RoutingTable only needs RaftPeerId.  So we can have List<RaftPeerId> instead of List<RaftPeer>.